### PR TITLE
TileMapHolder in MapEditorScreen is zoomable

### PR DESCRIPTION
**Pull request by r3versi**

### CODE REVIEW

**Comment by yairm210** [comment]
Looks good, but you've created a second class with the same name as, and has some common functionality with, the original TileMapHolder
Please put all common functionality (tile rendering, scroll, zoom) into a common class TileMapHolder, with specific inherited classes for each of the screens, like WorldScreenTileMapHolder and MapEditorTileMapHolder (because I'm feeling imaginative with names)

**Comment by r3versi** [comment]
Should i place the common `TileMapHolder` in `ui.utils`?

**Comment by yairm210** [comment]
Sounds good!

**Comment by yairm210** [comment]
Wow, the painting logic is good! I would not have thought of using a listener like that!

File: core/src/com/unciv/logic/HexMath.kt [hunk 1]
```diff
@@ -16,6 +14,18 @@ class HexMath {
...
     }
 
+    fun getAdjacentVectors(origin: Vector2): ArrayList<Vector2> {
+        val vectors = ArrayList<Vector2>()
+        vectors += Vector2(1f, 0f)
+        vectors += Vector2(1f, 1f)
+        vectors += Vector2(0f, 1f)
+        vectors += Vector2(-1f, 0f)
+        vectors += Vector2(-1f, -1f)
+        vectors += Vector2(0f, -1f)
```
Comment (naivekook) [inline core/src/com/unciv/logic/HexMath.kt]: this is already merged, but this is more kotlin way
`val vectors = listOf(
                Vector2(1f, 0f),
                Vector2(1f, 1f),
                Vector2(0f, 1f),
                Vector2(-1f, 0f),
                Vector2(-1f, -1f),
                Vector2(0f, -1f)
        )`

Comment (r3versi) [reply core/src/com/unciv/logic/HexMath.kt]: I'm going to add this to the refactor of HexMath. Thanks

**Pull request merged**